jetty-http2/http2-hpack/src/main/java/org/eclipse/jetty/http2/hpack/Huffman.java

```diff
@@ -488,4 +517,40 @@ private static void encode(final int[][] table, ByteBuffer buffer, String s)
             buffer.put((byte)(current));
         }
     }
+
+    private static void encode(final int[][] table, ByteBuffer buffer, byte[] b)
+    {
+        long current = 0;
+        int n = 0;
+
+        byte[] array = buffer.array();
+        int p = buffer.arrayOffset() + buffer.position();
+
+        int len = b.length;
+        for (int i = 0; i < len; i++)
+        {
+            int c = 0xFF & b[i];
+            int code = table[c][0];
+            int bits = table[c][1];
+
+            current <<= bits;
+            current |= code;
+            n += bits;
+
+            while (n >= 8)
+            {
+                n -= 8;
+                array[p++] = (byte)(current >> n);
+            }
+        }
+
+        if (n > 0)
+        {
+            current <<= (8 - n);
+            current |= (0xFF >>> n);
+            array[p++] = (byte)current;
+        }
+
+        buffer.position(p - buffer.arrayOffset());
+    }
```

Comment:
This whole method can be rewritten using standard ByteBuffer APIs without assuming it's a _heap_ ByteBuffer and has an array.
In light of #3952/#3959 would be great if we can limit the assumption that the ByteBuffer is heap based.